pkg/csi/service/wcp/controller.go

```diff
@@ -325,38 +325,40 @@ func (c *controller) createBlockVolume(ctx context.Context, req *csi.CreateVolum
 		} else if param == common.AttributeStoragePool {
 			storagePool = req.Parameters[paramName]
 			if !isValidAccessibilityRequirement(topologyRequirement) {
-				return nil, status.Errorf(codes.InvalidArgument, "invalid accessibility requirements")
+				return nil, logger.LogNewErrorCode(log, codes.InvalidArgument,
+					"invalid accessibility requirements")
 			}
 			spAccessibleNodes, storagePoolType, err := getStoragePoolInfo(ctx, storagePool)
 			if err != nil {
-				msg := fmt.Sprintf("Error in specified StoragePool %s. Error: %+v", storagePool, err)
-				return nil, status.Errorf(codes.Internal, msg)
+				return nil, logger.LogNewErrorCodef(log, codes.Internal,
+					"Error in specified StoragePool %s. Error: %+v", storagePool, err)
 			}
 			overlappingNodes, err := getOverlappingNodes(spAccessibleNodes, topologyRequirement)
 			if err != nil || len(overlappingNodes) == 0 {
-				msg := fmt.Sprintf("getOverlappingNodes failed: %v", err)
-				return nil, status.Errorf(codes.Internal, msg)
+				return nil, logger.LogNewErrorCodef(log, codes.Internal,
+					"getOverlappingNodes failed: %v", err)
 			}
 			accessibleNodes = append(accessibleNodes, overlappingNodes...)
 			log.Infof("Storage pool Accessible nodes for volume topology: %+v", accessibleNodes)
 
 			if storagePoolType == vsanDirect {
 				selectedDatastoreURL, err = getDatastoreURLFromStoragePool(ctx, storagePool)
 				if err != nil {
-					msg := fmt.Sprintf("Error in specified StoragePool %s. Error: %+v", storagePool, err)
-					log.Error(msg)
-					return nil, status.Errorf(codes.Internal, msg)
+					return nil, logger.LogNewErrorCodef(log, codes.Internal,
+						"error in specified StoragePool %s. Error: %+v", storagePool, err)
 				}
 				log.Infof("Will select datastore %s as per the provided storage pool %s", selectedDatastoreURL, storagePool)
 			} else if storagePoolType == vsanSna {
 				// Query API server to get ESX Host Moid from the hostLocalNodeName
 				if len(accessibleNodes) != 1 {
-					return nil, status.Errorf(codes.Internal, "too many accessible nodes")
+					return nil, logger.LogNewErrorCode(log, codes.Internal,
+						"too many accessible nodes")
 				}
 				hostMoid, err := getHostMOIDFromK8sCloudOperatorService(ctx, accessibleNodes[0])
 				if err != nil {
 					log.Error(err)
```

Comment:
Did you miss removing the log.Error here?

Reply:
Good eyes, Sandeep. The original code does not seem to follow the same pattern as the rest of code, i.e. 'err' is left out of the error message. So, I missed it.

log.Error(err)
return nil, status.Errorf(codes.Internal, "failed to get ESX Host Moid from API server")

Fixed.